### PR TITLE
group -e- add: splitting files by project and saving files to Drive

### PR DESCRIPTION
**Pull request by steeleb**
Heya Katie -

Last PR in this repo for a minute. Hopefully until you come back from your Japan trip!!

This group of functions breaks down the point pulls into our functional research groups: all CLP data ("CLP"), the subset of our lakes of interest ("ROSS_CLP"), and the northern water subset ("NW"). Note that this does not include polygon pulls at this time, as we have not established best practices for applying correction coefficients to those data. It is probably the same, but I need to think hard about that before I add them to the Drive folder. 

I'm currently working on populating a readme in the Drive folder where these files are stored. 

Pay special attention to any "DSWE1" or "point" hardcoding if it is not mentioned in the function name, please! Also of note, I had to add a few functions to group -a- in order to make this happen (basically to pull out the ROSS CLP data).

As always give me shout if you have any questions - since this is the last PR for this repo, no huge time crunch, as long as it's reviewed (and I've edited as necessary) before I leave next Friday I'll be happy!

B

### CODE REVIEW

File: e_separate_NW_CLP_data/src/add_spatial_information.R [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  # if data type is point, these will map 1:1 with row id
+  if (data_type == 'point') {
+    # left join with spatial info
+    data <- read_feather(data_file) %>% 
+      mutate(rowid = as.numeric(rowid)) %>% 
+      left_join(., spatial_info)
+  } 
+  # however, if it's a polygon, the rowid in the spatial info is the equivalent 
+  # of rowid + 1 in the RS data, since those data are pulled using python, which 
+  # uses base 0 instead of base 1
```
Comment (kathryn-willi) [inline e_separate_NW_CLP_data/src/add_spatial_information.R:35]: annoying -__-

File: e_separate_NW_CLP_data/src/subset_file_by_data_group.R [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#' @param data_grp character string; in this case, "NW" or "CLP", but could be any
+#' data group specified in group -a-
+#' 
+#' @returns filepath of resulting .feather file
+#' 
+#' 
+subset_file_by_data_group <- function(data_file, data_grp) {
+  data <- read_feather(data_file)
+  subset <- data %>% 
+    filter(data_group == !!data_grp)
```
Comment (kathryn-willi) [inline e_separate_NW_CLP_data/src/subset_file_by_data_group.R:17]: I've oddly never seen this before. what is the benefit of the `!!` ?

**Review by kathryn-willi — APPROVED** [review]

File: e_separate_NW_CLP_data/src/add_spatial_information.R (hunk 1 above)
Comment (steeleb) [reply e_separate_NW_CLP_data/src/add_spatial_information.R:35]: Truly. Moslty just annoying that I constantly have to check and re-check. lols.

File: e_separate_NW_CLP_data/src/subset_file_by_data_group.R (hunk 2 above)
Comment (steeleb) [reply e_separate_NW_CLP_data/src/subset_file_by_data_group.R:17]: Apparently it's called an 'injector' - and often works similarly to `{{ data_grp }}`, but not always. I'll be honest that I usually just try {{ var }} and !! var as necessary to get the expected output from a function... It also has something to do with the difference between 'symbolics' and 'quosures' which I truly don't understand. 

This is, by all means, and absolutely terrible non-explanation of your perfectly good question. If I manage to get a better grasp, I'll most certainly let you know. :)

**Comment by steeleb** [comment]
Thanks for jamming on this last week, KW. I appreciate you!

**Pull request merged**